src/arrayMethodSort.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let sort = compareFunction;
+
+    if (!sort) {
+      sort = function(a, b) {
+        if (String(a) > String(b)) {
+          return 1;
+        } else if (String(a) < String(b)) {
+          return -1;
+        }
+
+        return 0;
+      };
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
```

Comment:
let's think about the case when the array becomes sorted by less amount of iterations. Then the loop will make iterations without any changes. let's use `do-while` instead of the outer `for `and improve the algorithm complexity